### ISSUE TITLE
use -proto output instead of raw pprof output

### ISSUE THREAD

**Issue by irfansharif** [uber/go-torch#64]
Use of raw pprof output was introduced in #29.
The lack of a documented output grammar for pprof raw output has potential for an incomplete parser implementations, see #61 & #63 for two examples of this. Only publicly available 'grammar document' is [here](https://github.com/gperftools/gperftools/blob/855b3800064db49af823b85a54be269923eb6f4d/docs/cpuprofile-fileformat.html), from 2007.

Comments from #63 that point to why `-proto` output seems a more appropriate basis:
> I think https://github.com/google/pprof/blob/master/profile/profile.go#L431 is the best you're gonna get - note also that this points out that the raw format is "mainly for debug purposes". It may be the wrong format to ingest in the first place since it can probably change without notice and then it really matters which pprof version you run...

> The `-proto` flag seems more appropriate. See https://github.com/google/pprof/blob/master/profile/profile.go#L299 and https://github.com/google/pprof/blob/master/proto/profile.proto. But.. that's more work than just patching the raw format, except we wouldn't have to do it again over and over. 

-- @tschottdorf

**Comment by sandlerben** [comment]
I'll look into what work this would require

**Comment by sandlerben** [comment]
Ok, at first glance it seems like all the data from the raw output is in [the proto](https://github.com/google/pprof/blob/master/proto/profile.proto).

I think we could switch from parsing the raw output to reading directly from the proto without too much trouble. It would definitely be a non-trivial change, though. And we would have to test this new parser a lot. 

@prashantv what do you think about this?

I would love to work on this, but I'm not sure if I'd have the time to do it for quite a while.

**Comment by irfansharif** [comment]
re: testing, a lot of the fuzz testing applied to the google/pprof parser is applicable here. See https://github.com/google/pprof/tree/master/fuzz.

**Comment by prashantv** [comment]
I agree with moving to the proto, if we can do it soon. If it'll take us a while, it might not be worth doing as the pprof team is considering adding flamegraph output as an option directly:
https://github.com/google/pprof/issues/166
